### PR TITLE
[SCB-2245]upgrade netty to 4.1.60 and vert.x to 4.0.3

### PR DESCRIPTION
**Pull request by kaister3**

### CODE REVIEW

**Comment by yhs0092** [comment]
你好, 如果要升级Netty的话能否直接升级到`4.1.62.Final`呢? 60版本仍然是有漏洞的, 这个issue里面提到了  https://github.com/apache/servicecomb-java-chassis/issues/2299

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/SharedVertxFactory.java [hunk 1]
```diff
@@ -21,7 +21,9 @@
 
 import io.vertx.core.Vertx;
 import io.vertx.core.VertxOptions;
+import io.vertx.core.impl.VertxBuilder;
 import io.vertx.core.shareddata.Shareable;
+import io.vertx.core.spi.VertxMetricsFactory;
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/SharedVertxFactory.java:26]: unnecessary imports

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultVertxMetrics.java [hunk 2]
```diff
@@ -44,8 +46,10 @@
 
   public DefaultVertxMetrics(VertxOptions vertxOptions) {
     this.vertxOptions = vertxOptions;
+    MetricsOptions o = vertxOptions.getMetricsOptions();
+    System.out.println(o.getClass());
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultVertxMetrics.java:50]: remove this

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultTcpClientMetrics.java [hunk 3]
```diff
@@ -61,7 +61,6 @@ public void exceptionOccurred(DefaultTcpSocketMetric socketMetric, SocketAddress
 
   }
 
-  @Override
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultTcpClientMetrics.java:64]: seems remove override may have potential problems

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/MyTest.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package org.apache.servicecomb.foundation.vertx;
+
+import org.junit.Test;
+
+public class MyTest {
+  @Test
+  public void myTest() {
+    System.out.println(System.getProperty("java.io.tmpdir"));
```
Comment (liubao68) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/MyTest.java:8]: remove temporary code

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/client/tcp/TestNetClientWrapper.java [hunk 5]
```diff
@@ -78,18 +78,17 @@ public void connect(@Mocked NetSocket normalSocket, @Mocked NetSocket sslSocket)
     int port = 8000;
     String host = "localhost";
 
-    FutureFactoryImpl futureFactory = new FutureFactoryImpl();
     new MockUp<NetClient>(normalNetClient) {
       @Mock
       NetClient connect(int port, String host, Handler<AsyncResult<NetSocket>> connectHandler) {
-        connectHandler.handle(futureFactory.succeededFuture(normalSocket));
+        connectHandler.handle(new SucceededFuture<>(normalSocket));
```
Comment (wujimin) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/client/tcp/TestNetClientWrapper.java:84]: io.vertx.core.Future#succeededFuture(T)

**Pull request closed without merging**